lib/SIL/GraphFunctionDeviceInfo.cpp

```diff
@@ -125,22 +173,24 @@ bool GraphFunctionDeviceInfo::isConfigOp(const GraphOperationInfo &opInfo) {
          opInfo.getOperationName() == "tfc.configureCPU";
 }
 
-std::string
-GraphFunctionDeviceInfo::handleDevicePlacement(StringRef opType,
-                                               StringRef opDevice) {
+std::string GraphFunctionDeviceInfo::handleDevicePlacement(
+    StringRef opType, StringRef opDevice,
+    llvm::ArrayRef<GraphOperationAttribute> attributes) {
   DeviceType chosenDevice;
   if (!opDevice.empty())
     chosenDevice = getOpDeviceType(opDevice);
   else
-    chosenDevice = chooseDevice(opType);
+    chosenDevice = chooseDevice(opType, attributes);
```

Comment:
would it be useful to use LLVM_DEBUG to log the chosen device, here or within `chooseDevice()`?